indico/util/user.py

```diff
@@ -153,3 +155,25 @@ def wrapper(*args, **kwargs):
         return fn(*args, **kwargs)
 
     return wrapper
+
+
+def create_pending_user(user_data):
+    from indico.modules.users import User
+    from indico.modules.users.models.users import UserTitle
+
+    new_user = User()
+    new_user.email = user_data.get('email').lower()
+    new_user.first_name = user_data.get('firstName')
```

Comment:
no... just a plain function that takes the ugly dict and calls `create_pending_user` with proper arguments

or even easier, just take the values from the dict and pass them as normal arguments to `create_pending_user`..